backend/de.metas.material/dispo-commons/src/main/java/de/metas/material/dispo/commons/repository/CandidateRepositoryWriteService.java

```diff
@@ -977,4 +973,64 @@ public static class DeleteResult
 		@NonNull
 		BigDecimal previousQty;
 	}
+
+	public void updateQtyDetails(@NonNull final Candidate candidate, @NonNull final Candidate stockCandidate, @Nullable final Candidate previousStockCandidate)
+	{
+		final Candidate actualPreviousStockCandidate = CoalesceUtil.coalesceSuppliers(() -> previousStockCandidate,
+				() -> getPreviousStockCandidateOrNull(stockCandidate));
+		final BigDecimal previousQty = actualPreviousStockCandidate == null ? BigDecimal.ZERO : actualPreviousStockCandidate.getQuantity();
+		final CandidateId actualPreviousStockCandidateId = actualPreviousStockCandidate == null ? null : actualPreviousStockCandidate.getId();
+		final CandidateId currentCandidateId = candidate.getId();
+
+		final CandidateQtyDetailsPersistMultiRequest.CandidateQtyDetailsPersistMultiRequestBuilder builder = CandidateQtyDetailsPersistMultiRequest.builder()
+				.candidateId(currentCandidateId)
+				.stockCandidateId(stockCandidate.getId());
+		builder.details(ImmutableList.of(CandidateQtyDetailsPersistRequest.builder()
+						.detailCandidateId(actualPreviousStockCandidateId)
+						.qtyInStockUom(previousQty)
+						.build(),
+				CandidateQtyDetailsPersistRequest.builder()
+						.detailCandidateId(currentCandidateId)
+						.qtyInStockUom(getQuantityForCandidate(candidate))
+						.build()
+		));
+		candidateQtyDetailsRepository.save(builder.build());
+	}
+
+	private static BigDecimal getQuantityForCandidate(final @NonNull Candidate candidate)
```

Comment:
hmmm, reading the method name is not quite clear what kind in of quantity this method is returning and for which particular purpose
(actually, the mehtod name implies it's something general but reading the impl, it's not actually...)